diff --git a/ApplicationExeCode/CMakeLists.txt b/ApplicationExeCode/CMakeLists.txt
index f72f4518e..946c5e2a7 100644
--- a/ApplicationExeCode/CMakeLists.txt
+++ b/ApplicationExeCode/CMakeLists.txt
@@ -468,19 +470,6 @@ endforeach()
 # Installation packaging
 # ##############################################################################

-if(Qt6Widgets_VERSION VERSION_LESS 6.5.0)
-  message("Detected Qt version ${Qt6Widgets_VERSION}")
-  message(
-    "Install target is not complete as Qt 6.5 is required to use qt_generate_deploy_app_script()"
-  )
-else()
-  qt_generate_deploy_app_script(
-    TARGET ResInsight OUTPUT_SCRIPT deploy_script NO_UNSUPPORTED_PLATFORM_ERROR
-    NO_TRANSLATIONS
-  )
-  install(SCRIPT ${deploy_script})
-endif()
-
 if(${CMAKE_SYSTEM_NAME} MATCHES "Linux")
   set(CPACK_GENERATOR TGZ)
 elseif(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
