diff --git a/ThirdParty/custom-surfio/surfio/src/lib/irap_import_ascii.cpp b/ThirdParty/custom-surfio/surfio/src/lib/irap_import_ascii.cpp
index 4a5cf3c..fc2b5af 100644
--- a/ThirdParty/custom-surfio/surfio/src/lib/irap_import_ascii.cpp
+++ b/ThirdParty/custom-surfio/surfio/src/lib/irap_import_ascii.cpp
@@ -5,6 +5,8 @@
 #include <filesystem>
 #include <format>

+#include <fast_float/fast_float.h>
+
 namespace fs = std::filesystem;

 namespace surfio::irap {
@@ -15,7 +17,7 @@ auto is_space = [](char ch) { return facet.is(std::ctype_base::space, ch); };
 template <typename T, typename... U>
 const char* read_headers(const char* start, const char* end, T& arg, U&... args) {
   start = std::find_if_not(start, end, is_space);
-  auto [ptr, ec] = std::from_chars(start, end, arg);
+  auto [ptr, ec] = fast_float::from_chars(start, end, arg);
   if (ec != std::errc{})
     throw std::domain_error("Failed to read irap headers");

@@ -67,7 +69,7 @@ std::vector<float> get_values(const char* start, const char* end, int ncol, int
           )
       );

-    auto result = std::from_chars(start, end, value);
+    auto result = fast_float::from_chars(start, end, value);
     start = result.ptr;
     if (result.ec != std::errc())
       throw std::domain_error("Failed to read values during Irap ASCII import.");
diff --git a/ThirdParty/custom-surfio/CMakeLists.txt b/ThirdParty/custom-surfio/CMakeLists.txt
index 017dcd64e..6e1bfc7ce 100644
--- a/ThirdParty/custom-surfio/CMakeLists.txt
+++ b/ThirdParty/custom-surfio/CMakeLists.txt
@@ -27,3 +27,5 @@ target_include_directories(
   PRIVATE ../mio
 )

+find_package(FastFloat CONFIG REQUIRED)
+target_link_libraries(surfio_lib PRIVATE FastFloat::fast_float)
